api/edge/v1alpha1/types.go

```diff
@@ -171,6 +173,133 @@ type PlacementList struct {
 	Items           []Placement `json:"items"`
 }
 
+/******************** End Placement Type *********************/
+
+/******************** Start PlacementDecision Type *********************/
+
+// PlacementDecision is mapped 1:1 to a single Placement resource.
+// The decision resource reflects the resolution of the Placement's selectors,
+// and explicitly reflects which resources should go to what destinations.
+//
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster,shortName={pd}
+type PlacementDecision struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// `spec` explicitly describes a desired binding between workloads and Locations.
+	// It reflects the resolution of a Placement's selectors.
+	Spec PlacementDecisionSpec `json:"spec,omitempty"`
+
+	// `status` describes the status of the process of binding workloads to Locations.
```

Comment:
The described status is not the status of a "decision", it is the status of the _distribution_ . Something needs to be renamed or refactored here.

Reply:
how would you call it?

Reply:
Note also that propagation status has several facets. Propagation from WDS to mailbox. Propagation from mailbox to agent in WEC. Success or failure of agent to create/update object in WEC.

My thinking has been to (A) define something that exposes all these details in the mailbox and (B) let the user define summarization(s) of the aspects in the way that works best for them.

Reply:
this is the common k8s usage. status part reports the current state while spec part is the desired state.

Reply:
you're right about the several facets.
in this PR the facet is propagation from WDS to mailbox. 
this is what the transport controller is responsible for.
from the mailbox the objects are taken care by the agent.
we do plan on adding the other status of propagation from mailbox to WEC, but that is not part of this PR.
 
 please note that this PR do NOT cover the whole functionality of pluggable-transport. it just gives the starting point for phase1 as it was discussed with the broader team.
 
 I agree with you on both points (A) and (B), but both of them are out of scope of this PR. adding the status of mb to WEC propagation is more straight forward and can be added easily in an additional PR after this one is merged.
 the idea was to split PRs into smaller and easier to review PRs rather than one huge PR.
  
 regarding status and/or summarization, this should be designed more carefully before adding anything. 
 this topic was clearly agreed by the broader team and executives as out of scope for current release.

Reply:
btw, I documented that the status is for propagation from wds to mailbox in comments:
https://github.com/kubestellar/kubestellar/blob/transport-ks0.20/api/edge/v1alpha1/types.go#L279-#L280

Reply:
I also think that we should replace the one monolithic `Spec PlacementDecisionSpec` with two collections of slices (where decision slices and what decision slices) API objects. If we take the Spec out of the PlacementDecision object and leave WDS-to-mailbox propagation status then it clearly is no longer a placement _decision_ object.

For the WDS-to-mailbox propagation status, think also about overlap between Placement objects.

Reply:
There is no WDS-to-mailbox propagation status returned in release-0.14, and not currently in `main` either (if I understand correctly). Maybe we can separate the work on that status from the initial pluggable transport work.

Reply:
I'm not sure I understand the point of the last comment.
The design is completely different than what it was in release 0-14, obviously there are many different things + things that were not part of previous releases. same goes for current main.

Reply:
regarding the point about replacing `Spec` - I disagree with the comment.
every CR has spec and status fields, this is the k8s way of defining CRs. we don't want to re-invent the wheel.
we use the k8s common patterns by defining spec and status fields and under the spec you can find the two collections you specified. these two collections represent the desired state of this CR (desired state in k8s CRs is what's under spec, not an arbitrary definition of two collections).

Reply:
An object having spec and status is the common pattern but not the only pattern. For examples in Kubernetes: ConfigMap, Secret, Endpoints, EndpointsSlice. The last one is an inspiration for what we did about the "where" resolution in the release 0.15 and prior, and the extension I proposed in #1222 . In the world of "compliance" and "policy" it is not uncommon to have reports in different objects than policies. See the work of the Compliance-to-Policy project for an example.

I think there is design work still to do on status reporting. I suggested leaving the status reported out of the initial pluggable transport work so that the status reporting design work not hold up the other initial pluggable transport work.

Reply:
I'm OK leaving the status fields out of this PR.
let me update the api definition quickly.

Reply:
done